python/plugins/processing/algs/gdal/ClipRasterByExtent.py

```diff
@@ -161,7 +162,32 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
             extra = self.parameterAsString(parameters, self.EXTRA, context)
             arguments.append(extra)
 
-        arguments.append(inLayer.source())
+        layerSource = inLayer.source()
+        layerProvider = inLayer.dataProvider()
+        if (layerProvider.name() == 'postgresraster'):
+            metadata = layerProvider.htmlMetadata()
+            uri_config = {}
+            tiled = re.search(r"Is Tiled</td><td>([^<]+)", metadata)
```

Comment:
@elpaso is there a better way to handle this? This approach looks extremely fragile to me

Reply:
Fully agree. A more robust approach has to expose some of the properties like `isTiled` to choose mode 1 or 2. @elpaso comments are very well welcome.

Reply:
There is no API to retrieve that information. If you want to add it it would part of `QgsRasterDataProvider`.

The same goes for the `gdalSource()`.

But my main question is: why don't you use QGIS API for the clip algorithm instead of proxying to GDAL? 

This would work with any raster supported by QGIS.

Reply:
> But my main question is: why don't you use QGIS API for the clip algorithm instead of proxying to GDAL?
> 

@elpaso Thanks for the feedback. I'm not aware of such API. Can you provided me a search term for that?

Reply:
The building blocks should be available: https://qgis.org/api/classQgsRasterInterface.html#a0f28ed600d4fabd86c88c520652b8ae6 and the raster calculator can probably handle the calculations.

I'm not sure all the pieces are available, but in general I think that using QGIS API for raster operations would be a huge step forward (you can still use GDAL for the low level operations, no need to reinvent the wheel).

Reply:
I agree with you @elpaso, but for addressing this issue (which is a regression in the sense that clipping was working with any raster layers until now), I would go for a easier solution. To proper define and expose even a few raster operations is not trivial.

I've called it **band-aid** fix exactly because I would much better if we have a well defined implementation of some raster operations. 

Do you think it is acceptable to fix this almost as it is (plus exposing a `tiled` flag to `QgsRasterDataProvider` to avoid the fragile call to `htmlMetadata`)? If not, I'll have to investigate further how to add a `clipped` method to `QgsRasterInterface`, either based on a `QgsRectangle` or on a `QgsVectorLayer`.

Either way, adding a `tiled` flag to `QgsRasterDataProvider` seems useful. Do you agree? Should this go to `QgsRasterInterface` with `false` as default or keep it just for the `QgsRasterDataProvider` class?

Reply:
> The building blocks should be available: https://qgis.org/api/classQgsRasterInterface.html#a0f28ed600d4fabd86c88c520652b8ae6 and the raster calculator can probably handle the calculations.
> 
> I'm not sure all the pieces are available, but in general I think that using QGIS API for raster operations would be a huge step forward (you can still use GDAL for the low level operations, no need to reinvent the wheel).

I'll follow this approach, which seems more interesting for the long term. This PR as it is can be used for a personal fix meanwhile. I'll close this PR.